nano 4.6 fails to build on macOS

**Describe the bug**
nano fails to build:

~~~
clang -DHAVE_CONFIG_H -I. -I..  -DLOCALEDIR=\"/nix/store/x9a0w6f62vj6xx4ip6mhnpzbyixcdxhn-nano-4.6/share/locale\" -DSYSCONFDIR=\"/etc\" -I../lib -I../lib   -g -O2 -Wall -c -o nano.o nano.c
text.c:2529:33: error: no member named 'st_mtim' in 'struct stat'
        timestamp_sec = (long)fileinfo.st_mtim.tv_sec;
                              ~~~~~~~~ ^
text.c:2530:34: error: no member named 'st_mtim' in 'struct stat'
        timestamp_nsec = (long)fileinfo.st_mtim.tv_nsec;
                               ~~~~~~~~ ^
text.c:2570:21: error: no member named 'st_mtim' in 'struct stat'
        if ((long)fileinfo.st_mtim.tv_sec == timestamp_sec &&
                  ~~~~~~~~ ^
text.c:2571:20: error: no member named 'st_mtim' in 'struct stat'
                                (long)fileinfo.st_mtim.tv_nsec == timestamp_nsec) {
                                      ~~~~~~~~ ^
4 errors generated.
make[2]: *** [Makefile:1618: text.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/private/tmp/nix-build-nano-4.6.drv-0/nano-4.6/src'
make[1]: *** [Makefile:1577: all-recursive] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-nano-4.6.drv-0/nano-4.6'
make: *** [Makefile:1516: all] Error 2
builder for '/nix/store/x9vh1424xcx71bpiz4b86klij1yg3qi7-nano-4.6.drv' failed with exit code 2
~~~

**To Reproduce**
Steps to reproduce the behavior:
1. Use macOS
2. Update to `master`
3. `nix-env -i nano`

**Expected behavior**
Build nano.

**Metadata**
Please run `nix run nixpkgs.nix-info -c nix-info -m` and paste the result.

 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.0.0, macOS 10.15.1`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.1`
 - channels(crestani): `"nixpkgs-19.03pre153206.f753852e11d"`
 - nixpkgs: `/Users/crestani/nixpkgs`

--- a/pkgs/applications/editors/nano/default.nix
+++ b/pkgs/applications/editors/nano/default.nix
@@ -38,6 +38,15 @@ in stdenv.mkDerivation rec {
     (stdenv.lib.enableFeature enableTiny "tiny")
   ];
 
+    patches = [
+      (fetchurl {
+        # fix compilation on macOS, where 'st_mtim' is unknown
+        # upstream patch not in 4.6
+        url = "https://git.savannah.gnu.org/cgit/nano.git/patch/?id=f516cddce749c3bf938271ef3182b9169ac8cbcc";
+        sha256 = "0gqymvr5vxxypr7y3sm252rsi4gjqp597l01x0lkxyvxsn45a4sx";
+      })
+  ];
+
   postInstall = ''
     cp ${nixSyntaxHighlight}/nix.nanorc $out/share/nano/
   '';

